File: SpringBoot/AnotherProject/controllers/UsersController.java
package com.richard.estudos.anotherproject.controllers;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RestController;

import com.richard.estudos.anotherproject.controller.requests.UserRequest;

@RestController
@RequestMapping(value="/users")
public class UsersController {

	@RequestMapping(method=RequestMethod.GET, value="/")
	public ResponseEntity<String> list() {
		return new ResponseEntity<String>(HttpStatus.OK);
	}
	
	@RequestMapping(method=RequestMethod.POST, value="/")
	public ResponseEntity<String> create(@RequestBody UserRequest request) {
		return new ResponseEntity<String>(HttpStatus.OK);
	}
}

File: Kotlin/dynkot/src/main/kotlin/com/richard/dynkot/controllers/UsersController.kt
package com.richard.dynkot.controllers

import com.richard.dynkot.controllers.requests.CreateUserRequest
import com.richard.dynkot.controllers.responses.UserResponse
import com.richard.dynkot.services.UsersService
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/users")
class UsersController(private val usersService: UsersService) {

    @PostMapping
    fun create(createUserRequest : CreateUserRequest) {

    }

    @GetMapping("/{email}")
    fun findByEmail(@PathVariable("email") email : String) : UserResponse {
        val user = usersService.findByEmail(email)
        return UserResponse(user.email, user.name)
    }
}